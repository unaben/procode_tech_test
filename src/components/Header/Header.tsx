import React from "react";
import { message } from "../../translation/ENT";
import './Header.css'


const Header = () => {
  return <h1>{message.myTodo}</h1>;
};

export default Header;
