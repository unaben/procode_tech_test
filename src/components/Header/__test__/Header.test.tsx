import { render, screen } from "@testing-library/react";
import Header from "../Header";
import { Provider } from "react-redux";
import { store } from "../../../app/store";

describe("Header component", () => {
  it("should render correctly", () => {
    const { container } = render(
      <Provider store={store}>
        <Header />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it("should render heading element", () => {
    render(
      <Provider store={store}>
        <Header />
      </Provider>
    );
    const headingEl = screen.getByRole("heading", { level: 1 });
    expect(headingEl).toBeInTheDocument();
    expect(headingEl).toHaveTextContent(/Basic Todo App/i);
  });
});
