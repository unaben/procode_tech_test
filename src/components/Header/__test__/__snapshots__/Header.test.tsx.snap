// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly 1`] = `
<div>
  <h1>
    TODOs
  </h1>
</div>
`;
